### DISCUSSION TITLE
Unable to obtain the sysname of the switch, but it can be obtained through snmpwalk

### DISCUSSION THREAD

**Discussion by aliuzzz** [prometheus/snmp_exporter#975] [Q&A]
![image](https://github.com/prometheus/snmp_exporter/assets/43594093/56932abc-863a-4eb8-8509-514260b985ac)
this is my snmp.yml，I have used a new template, sysuptime can be obtained, and ifnumber can also be obtained normally, but sysname has been invalidated. What is the reason for this? My snmp_ Exporter startup did not see relevant error logs
![image](https://github.com/prometheus/snmp_exporter/assets/43594093/10e4d891-fa93-478b-a738-861cd4ec78b5)
![image](https://github.com/prometheus/snmp_exporter/assets/43594093/fe383ad2-8163-4608-94d4-2608ef99e24f)

**Comment by G0nz0uk** [comment]
Did you get to the bottom of this as?  I’m getting a similar issue with trying to add ‘sysName’
